### ISSUE TITLE
Global stacks stuck in Updating-Active after Rancher upgrade to 0.63.0-rc2

### ISSUE THREAD

**Issue by sshipway** [rancher/rancher#3922]
Rancher Version: 0.63.0-rc2
Docker Version: 1.9.1
OS: CoreOS
Steps to Reproduce: 
Running  rancher 0.59.1.  Existing running stack with global service instances.  Upgraded to Rancher 0.63.0-rc2.  Stack stuck in updating-active status "Failed to do service reconcile"

Results: 
Stack in status Updating-Active
   Updating-Active (Failed to do service reconcile for service [114])
However all service instances were still running correctly.
This seems to affect only the Global services for some reason.
Restarting the stack completely results in it being stuck in Activating status
   Activating (Object [instance:1i5628] failed to satisfy predicate [15000] millis. instance status: Networking)

Expected:
Working system with no different state

**Comment by deniseschannon** [comment]
Verified that this is occurring. Errors are being logged in rancher-server.

```
2016-03-11 04:55:54,017 ERROR [:] [] [] [] [cutorService-11] [o.a.c.m.context.NoExceptionRunnable ] Uncaught exception java.lang.IllegalStateException: Failed to do service reconcile for service [8]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl$1.doWithLock(DeploymentManagerImpl.java:135) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl$1.doWithLock(DeploymentManagerImpl.java:109) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl.activate(DeploymentManagerImpl.java:109) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl.activate(DeploymentManagerImpl.java:92) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl$4.run(DeploymentManagerImpl.java:306) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl.reconcileForClient(DeploymentManagerImpl.java:317) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.servicediscovery.deployment.impl.DeploymentManagerImpl.serviceUpdate(DeploymentManagerImpl.java:301) ~[cattle-iaas-service-discovery-server-0.5.0-SNAPSHOT.jar:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.7.0_95]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57) ~[na:1.7.0_95]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.7.0_95]
	at java.lang.reflect.Method.invoke(Method.java:606) ~[na:1.7.0_95]
	at io.cattle.platform.eventing.annotation.MethodInvokingListener$1.doWithLockNoResult(MethodInvokingListener.java:76) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:7) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.LockCallbackNoReturn.doWithLock(LockCallbackNoReturn.java:3) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl$3.doWithLock(AbstractLockManagerImpl.java:40) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.LockManagerImpl.doLock(LockManagerImpl.java:33) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:13) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.lock.impl.AbstractLockManagerImpl.lock(AbstractLockManagerImpl.java:37) ~[cattle-framework-lock-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.eventing.annotation.MethodInvokingListener.onEvent(MethodInvokingListener.java:72) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at io.cattle.platform.eventing.impl.AbstractThreadPoolingEventService$2.doRun(AbstractThreadPoolingEventService.java:135) ~[cattle-framework-eventing-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.NoExceptionRunnable.runInContext(NoExceptionRunnable.java:13) ~[cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable$1.run(ManagedContextRunnable.java:49) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:55) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:108) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:52) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at org.apache.cloudstack.managed.context.ManagedContextRunnable.run(ManagedContextRunnable.java:46) [cattle-framework-managed-context-0.5.0-SNAPSHOT.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_95]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_95]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_95]

```
In my case, the services were eventually active.

**Comment by alena1108** [comment]
@deniseschannon this error is generic, and doesn't give any details of why it fails...could you point me to the setup of yours? So I can look at the logs/db

**Comment by deniseschannon** [comment]
@sshipway In my upgraded setup, the services all eventually turned `Active`, but it took a little bit of time. It wasn't instantaneous. 

@alena1108 Could you describe why it goes into updating-active state?

**Comment by alena1108** [comment]
@deniseschannon no idea why it happens, sounds like a bug. Only new host addition or container dying should have caused it.

**Comment by sshipway** [comment]
On my setup, it seemed to affect only a few instances of certain services.  When I deleted those specific containers, the scheduler re-created them and it worked OK.

**Comment by alena1108** [comment]
@sshipway did your problematic services have health checks enabled?

**Comment by alena1108** [comment]
@sangeethah should be tested as part of next upgrade testing - make sure you have global services with health check enabled and disabled on pre-upgraded setup

**Comment by sangeethah** [comment]
@alena , is this bug not the one encountered in https://github.com/rancher/rancher/issues/3933 ?

**Comment by sangeethah** [comment]
There were 2 issues that were fixed relating to this issue:

UI bug - https://github.com/rancher/rancher/issues/3904 - where UI switched back to using UI calculated health states from using healthstate API for release 0.63.1.
Upgrade bug -  #3933  which was addressed in  0.63.1
